### PR TITLE
OSDOCS-773 Azure BYO VNet

### PR DESCRIPTION
**Pull request by kalexand-rh**
https://jira.coreos.com/browse/OSDOCS-773

@jhixson74, will you PTAL?

### CODE REVIEW

File: modules/installation-about-custom-azure-vnet.adoc [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+* The VNet and its subnets must belong to the same resource group, and the subnets must be configured to use Azure-assigned DHCP IP addresses instead of static IP addresses.
+
+You need to provide two subnets within your VNet, one for the control plane machines and one for the compute machines. Because Azure distributes machines in different availability zones within the region that you specify, your cluster will have high availability by default.
+
+To ensure that the subnets that you provide to the installation program are suitable, it confirms the following data:
+
+* All the subnets that you specify exist.
+* You provide private subnets.
+* The subnet CIDRs belong to the machine CIDR that you specified.
+* Unless you use a private cluster, you must provide exactly one public and private subnet for each availability zone. Each availability zone contains no more than one public and one private subnet.
```
Comment (jhixson74) [inline modules/installation-about-custom-azure-vnet.adoc]: The subnets provided to the install config for Azure are both private subnets. Public IP addresses are allocated for the load balancers that sit in front of the control plane and worker nodes.

File: modules/installation-about-custom-azure-vnet.adoc [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+////
+These are some of the details from the AWS version, and if any of them are relevant to Azure, they can be included.
+If you deploy {product-title} to an existing network, the isolation of cluster services is reduced in the following ways:
+
+* You can install multiple {product-title} clusters in the same VNet.
+* ICMP ingress is allowed to entire network.
+* TCP 22 ingress (SSH) is allowed to the entire network.
+* Control plane TCP 6443 ingress (Kubernetes API) is allowed to the entire network.
+* Control plane TCP 22623 ingress (MCS) is allowed to the entire network.
+////
```
Comment (jhixson74) [inline modules/installation-about-custom-azure-vnet.adoc]: Something else to be aware of, is the NSG's that are created for the subnets need to have rules to allow communication with the control plane, http, https, etc. See  https://github.com/openshift/installer/pull/2521#issuecomment-547190111 for the rules I used.

Comment (kalexand-rh) [reply modules/installation-about-custom-azure-vnet.adoc]: I'm adding a table that contains the ports that we discussed.

**Comment by kalexand-rh** [comment]
@jhixson74, I've made changes that I think address your VPC-related comments. Please let me know if you have any other comments, requests, or corrections.

@etienne, will you PTAL?

**Comment by e-tienne** [comment]
LGTM

**Comment by kalexand-rh** [comment]
@mgahagan73, Etienne said that you should probably take a look, too. Will you please let me know if you see any issues?

**Comment by kalexand-rh** [comment]
@mgahagan73, will you PTAL?

**Comment by mgahagan73** [comment]
LGTM

**Comment by kalexand-rh** [comment]
@openshift/team-documentation, PTAL?

File: installing/installing_azure/installing-azure-vnet.adoc [hunk 3]
```diff
@@ -0,0 +1,49 @@
+[id="installing-azure-vnet"]
+= Installing a cluster on Azure to an existing VNet
+include::modules/common-attributes.adoc[]
+:context: installing-azure-vnet
+
+toc::[]
+
+In {product-title} version {product-version}, you can install a customized
```
Comment (bergerhoffer) [inline installing/installing_azure/installing-azure-vnet.adoc]: Should this intro paragraph mention to an existing VNet?

File: modules/installation-about-custom-azure-vnet.adoc [hunk 4]
```diff
@@ -0,0 +1,109 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_azure/installing-azure-vnet.adoc
+
+[id="installation-about-custom-azure-vnet_{context}"]
+= About reusing a VNet for your {product-title} cluster
+
+In {product-title} {product-version}, you can deploy a cluster into an existing VNet in Microsoft Azure. If you do, you must also use existing subnets within the VNet and routing rules.
+
+By deploying {product-title} into an existing Azure VNet, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that your company's guidelines set. This is a good option to use if you cannot obtain the infrastructure creation permissions that are required to create the VNet yourself.
```
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: Same comment as other PR about picking either "avoid" or "limit"

File: modules/installation-about-custom-azure-vnet.adoc [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+[IMPORTANT]
+====
+Use of an existing VNet requires the use of the updated Azure Private DNS (preview) feature. See link:https://azure.microsoft.com/en-us/updates/announcing-preview-refresh-for-azure-dns-private-zones-2/[Announcing Preview Refresh for Azure DNS Private Zones] for more information about the limits of this feature.
```
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: Suggest limitations instead of limits in the last sentence.

Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: Suggest starting the sentence with the: "The use of an existing..."

File: modules/installation-about-custom-azure-vnet.adoc [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+====
+
+[id="installation-about-custom-azure-vnet-requirements_{context}"]
+== Requirements for using your VNet
+
+The installation program will no longer create the following components:
+* Subnets
```
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: Need an empty line above this - the bullets aren't rendering.

File: modules/installation-about-custom-azure-vnet.adoc [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+[IMPORTANT]
+====
+Use of an existing VNet requires the use of the updated Azure Private DNS (preview) feature. See link:https://azure.microsoft.com/en-us/updates/announcing-preview-refresh-for-azure-dns-private-zones-2/[Announcing Preview Refresh for Azure DNS Private Zones] for more information about the limits of this feature.
+====
+
+[id="installation-about-custom-azure-vnet-requirements_{context}"]
+== Requirements for using your VNet
+
+The installation program will no longer create the following components:
```
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: will no longer create -> no longer creates

Like in your AWS PR. Though I admit I don't understand what this means. Do you mean as compared to 4.2? Or it doesn't create these only in the even that you're using your own VNet (or VPC?)

File: modules/installation-about-custom-azure-vnet.adoc [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+The cluster must be able to access the resource group that contains the existing VNet and subnets. While all of the resources that the cluster creates are placed in a separate resource group that it creates, some network resources are used from a separate group. Some cluster Operators must be able to access resources in both resource groups. For example, the Machine API controller attaches NICS for the virtual machines that it creates to subnets from the networking resource group.
+
+Your VNet must meet the following characteristics:
+
+* The VNet’s CIDR block must contain the `Networking.MachineCIDR` range, which is the IP address pool for cluster machines.
+* The VNet and its subnets must belong to the same resource group, and the subnets must be configured to use Azure-assigned DHCP IP addresses instead of static IP addresses.
+
+You need to provide two subnets within your VNet, one for the control plane machines and one for the compute machines. Because Azure distributes machines in different availability zones within the region that you specify, your cluster will have high availability by default.
```
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: You need to -> You must

File: modules/installation-about-custom-azure-vnet.adoc [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+To ensure that the subnets that you provide to the installation program are suitable, it confirms the following data:
```
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: Same comment here as the other PR about rearranging the sentence to be clearer.

File: modules/installation-about-custom-azure-vnet.adoc [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+* All the subnets that you specify exist.
+* You provide two private subnets for each availability zone.
+* The subnet CIDRs belong to the machine CIDR that you specified. Machines are not be provisioned in availability zones that you do not provide private subnets for. If required, the installation program creates public load balancers that manage the control plane and worker nodes, and Azure allocates a public IP address to them.
```
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: are not be -> are not ?

File: modules/installation-about-custom-azure-vnet.adoc [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+.Required ports
+[options="header",cols="1,3,1,1"]
+|===
+
+|Port
+|Description
+|Control plane
+|Compute
+
+////
```
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: This isn't commented out, it's being rendered in the doc.

File: modules/installation-about-custom-azure-vnet.adoc (hunk 4 above)
Comment (bergerhoffer) [inline modules/installation-about-custom-azure-vnet.adoc]: Same comment as other PR about moving the word "yourself" to be clearer.

**Review by bergerhoffer — COMMENTED** [review]
A few things - some the same comments as the AWS PR. I tried to point things out that I saw in both, but I'd double check to make sure that I didn't miss any here that I pointed out in the AWS PR.

File: modules/installation-about-custom-azure-vnet.adoc (hunk 7 above)
Comment (kalexand-rh) [reply modules/installation-about-custom-azure-vnet.adoc]: Thank you for pointing out that this wasn't clear enough. I'm thinking of using this text instead:

> When you deploy a cluster by using an existing VNet, you must perform additional network configuration before you install the cluster. In installer-provisioned infrastructure clusters, the installer usually creates the following components, but it does not create them when you install into an existing VNet:

**Pull request merged**